### ISSUE TITLE
JoinFaces Reference Guide for primefaces example results in JSP version container to old, EL-RI Factory could not determined

### ISSUE THREAD

**Issue by StefanSchubert** [joinfaces/joinfaces#1386]
Hi there,

during upgrading from spring boot 2.4.3 to spring boot 2.6.3 following the reference guide I got errors on runtime
erros telling my that tomcats JSP version is to old and that the EL-RI Factory couldn't fetched.

Please check the reference example on joinfaces.org

To reproduce you may clone https://github.com/StefanSchubert/JSF-Springboot-Blueprint
You may see in the pom that I tryied to fiddle with providing the factory from 2011 glassfish dep, but this led to other problems.

I'm also a bit confused that the reference guide also for primefaces says to use jfs-spring-boot-starter.
Is the primefaces starter really included an discontinued?

Thanks for help in advance.
Stefan

**Comment by larsgrefer** [comment]
You have 2 conflicting el-apis in your classpath:

- `javax.el:javax.el-api` pulled by your cdi-api
- `org.apache.tomcat.embed:tomcat-embed-el` (pulled by spring-boot-starter-tomcat)

excluding the first one fixes the problem:
```xml
        <dependency>
            <groupId>javax.enterprise</groupId>
            <artifactId>cdi-api</artifactId>
            <version>2.0.SP1</version>
            <exclusions>
                <exclusion>
                    <groupId>javax.el</groupId>
                    <artifactId>javax.el-api</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

**Comment by larsgrefer** [comment]
When you switch to the jakarta cdi dependency you can inherit the exclusion from our dependencyManagement:

```xml
        <dependency>
            <groupId>jakarta.enterprise</groupId>
            <artifactId>jakarta.enterprise.cdi-api</artifactId>
        </dependency>
```

**Comment by StefanSchubert** [comment]
Hi Lars,

thanks a lot for Sunday support :-) 

I was investigating in parallel and comparing the example a bit closer, I also identified the cdi-api as the problem. 
While comparing I rewrote the example and get rid of the cdi-api and using springs default DI.
Just checked in the fixed example.

As I had the problem in my main sabi project too, I decided there to switch to your advised jakarta solution and it works there well. Saves me a lot of refactoring! 

So now I'm on spring boot 2.6.3 and primefaces 11 there.

THX again and have a good start into next week.
Stefan
